cmake/gpu.cmake

```diff
@@ -14,20 +14,23 @@ if(EXISTS ${FILEPATH_CURRENT_DIR})
 elseif(EXISTS ${FILEPATH_PROJECT_ROOT})
     set(TARGET_FILE_PATH ${PROJECT_ROOT})
 else()
-    message(FATAL_ERROR "File ${FILENAME} not found in either ${CMAKE_CURRENT_SOURCE_DIR} or ${CMAKE_CURRENT_SOURCE_DIR}/../../")
+    message(
```

Comment:
Would you mind replacing all this file finding logic in gpu.cmake and example.cmake with

```cmake
# Specify the filename to search for
set(FILENAME "gpu.h")

# Function to check for file existence up the directory hierarchy
function(find_project_root current_dir filename result_var)
    set(found FALSE)  # Flag to indicate if the file is found
    set(current_check_dir "${current_dir}")  # Start from the given directory
    
    # Windows flutter embeds 8 levels deep
    foreach(i RANGE 0 2)
        set(filepath "${current_check_dir}/${filename}")
        
        if(EXISTS "${filepath}")
            set(${result_var} "${current_check_dir}" PARENT_SCOPE)
            set(found TRUE)
            break()
        endif()
        
        # Move one level up
        get_filename_component(current_check_dir "${current_check_dir}" DIRECTORY)
    endforeach()
    
    if(NOT found)
        set(${result_var} "" PARENT_SCOPE)  # Set to empty if not found
    endif()
endfunction()
```

then use as 

```cmake
find_project_root("${CMAKE_CURRENT_SOURCE_DIR}" "${FILENAME}" TARGET_FILE_PATH)
```

Reply:
Done, still trying to put [this](https://github.com/jgh-/webgpu-workbench/blob/d5d505462e3b0c61a98211e9170b85e2e18941cc/cmake/dawn.cmake#L3) into the script